add generating shims for terraform 0.13 (#3)
numtide/generate-terraform-provider-shim: README.md

exec "${plugin_binary_path}" $@
```

> With release of Terraform 0.13.x there is a standardized way of distributing the provider binaries, but the burden of running a provider registry and signing provider binaries is on the provider developers. 

Reviewer: ```suggestion
With the release of Terraform 0.13.x there is a standardized way of distributing the provider binaries, but the burden of running a provider registry and signing provider binaries is on the provider developers. 
```

This means that most of the providers will not be immediately available through a registry.
For such providers, provider shims is a stopgap solution until community providers can be downloaded through a registry.

## Usage

running

```sh
$ generate-terraform-provider-shim numtide/terraform-provider-linuxbox
```

will generate provider shims for Linuxbox provider in the `.terraform/plugins/` and `terraform.d/plugins/registry.terraform.io/hashicorp/linuxbox/<version>` directories for each arch respectively.

## Terraform 0.13 support

At the moment, there are plenty of community providers that are not available in the Hashicorp's registry and are not running registries of their own.

> This will change with time and this tool will be less useful, but for the transition period, this tool will generate shims in the correct paths for both terraform 0.12 and 0.13.

Reviewer: ```suggestion
This will change with time and this tool will become less useful, but for the transition period, this tool will generate shims in the correct paths for both terraform 0.12 and 0.13.
```

Generated shims will let providers appear as if they were downloaded through Hashicorp's registry, so there is no need to add an entry in `terraform{ required_providers{} }` specifying the location of the provider registry.

## Limitations

### Only GitHub hosted providers

> Shim generator is integrating tightly with GitHub's API to find release archives, hence will be able to generate shims only for the providers that are available on GitHub.

Reviewer: ```suggestion
Shim generator is integrating tightly with GitHub's API to find release archives, hence will be able to generate shims only for the providers that are available on GitHub.

(we are open to accepting PRs to support other sources)
```

### Only providers with releases with attached binaries are supported

> We are relying on developers of the providers to create releases with attached compiled binaries of the providers for different archs.

Reviewer: ```suggestion
We are relying on the developers of the provider to create releases with attached compiled binaries of the providers for different architectures.
```

> If that is not the case, there shim cannot be generated.

Reviewer: ```suggestion
If that is not the case, the shim cannot be generated.
```

### Only .tar.gz and .zip archives are supported
There is no standard way of packaging providers.
Most of the time they are packaged in a Zip or Gzipped Tar archive - those are formats we are supporting.
If you would like us to add support for some other formats, please create an issue in this repo with a link to the specific provider you would like us to support.

### Windows is not supported

> We do not have access to a Windows machine and have never run terraform in Windows environment, hence the generated shims will definitely not work under Windows.

Reviewer: ```suggestion
We do not have access to a Windows machine and have never run terraform in a Windows environment, hence the generated shims will definitely not work under Windows.
```
